app/narwos.py

```diff
@@ -273,6 +277,52 @@ def show_clusters():
     
     return 'Nothing to show here'
 
+@app.route('/show_filters_submit', methods=['POST'])
+def show_filter_submit():
+
+    show_filter = request.form.get('show_filter')
+    if show_filter:
+        return redirect(url_for('show_filter', show_filter=show_filter))
+    else:
+        return redirect(url_for("index", message=f"Cannot show the filtering!"))
+
+@app.route('/show_filters', methods=['GET'])
+def show_filter():
+
+    df = pd.read_csv(PATH_TO_CURRENT_DF)
+
+    if isinstance(df, pd.DataFrame):
+
+        return render_template('filtering.html', columns=df.columns)
+    
+    return 'Nothing to show here'
+
+@app.route('/filter', methods=['POST'])
+def filter_data():
+    filters = request.get_json()
+    filtered_df = pd.read_csv(PATH_TO_CURRENT_DF)
+    filtered_df = filtered_df.astype(str)
+    logger.info(
+        f"""Columns of df to filtr:{filtered_df.columns}""")
+    for filter_data in filters:
+        column = filter_data['column']
```

Comment:
If you want to retrieve value from dict, use .get(), when you try to retrieve value for key which does not exists, you would get error, you have to service this, maybe add some logic with try, except if these filters are not found.